app/libs/utils.py

```diff
@@ -1,7 +1,7 @@
 def convert_tx_id(tx_id: str) -> str:
     """
-    Converts the guid tx_id to string of 16 characters with a dash between every 4 characters
+    Converts the guid tx_id to uppercase string of 16 characters with a dash between every 4 characters
     :param tx_id: tx_id to be converted
     :return: String in the form of xxxx-xxxx-xxxx-xxxx
     """
-    return (tx_id[:4] + "-" + tx_id[4:])[:19]
+    return " - ".join(((tx_id[:4] + "-" + tx_id[4:])[:19]).split("-")).upper()
```

Comment:
This can be done without splitting and joining by using `replace`:
```suggestion
    return ((tx_id[:4] + "-" + tx_id[4:])[:19]).replace("-", " - ").upper()
```